### DISCUSSION TITLE
Firebase 400 Error

### DISCUSSION THREAD

**Discussion by BenJarrett** [nss-evening-cohort-14/evening-client-side#189] [Help Me!] [answered]
Ben Jarrett

#168 Was similar, but it did not specifically pertain to my issue.

When I hit my submit pin button, my click event to show pins, and on page load I get this error
GET https://pinterest-assignment-new-default-rtdb.firebaseio.com/pins.json?orderBy=%22boards_Id%22&equalTo=%22%22 400 (Bad Request)
And
Uncaught (in promise) Error: Request failed with status code 400
at createError (createError.js?2d83:16)
at settle (settle.js?467f:17)
at XMLHttpRequest.handleLoad (xhr.js?b50d:62)

So I know these are Firebase errors.
When I searched the first error in PostMan, I get "error": "Index not defined, add ".indexOn": "boards_Id", for path "/pins", to the rules"
So I checked my firebase rules:
```
  "rules": {
    ".read": "now < 1617426000000",  // 2021-4-3
    ".write": "now < 1617426000000",  // 2021-4-3
      "boards": {
        ".indexOn": ["uid", "firebaseKey"]
      },
        "pins": {
        ".indexOn": ["uid", "firebaseKey", "boards_id"]
      },
  }
}
```
Not sure what exactly is wrong with my rules

Sorry for the Stack Overflow and Reddit links. There were very few results that actually pertained to this error when researching.

https://firebase.google.com/docs/reference/fcm/rest/v1/ErrorCode
https://www.reddit.com/r/Firebase/comments/gvrj5j/firebase_dynamic_links_rest_api_returns_400_bad/
https://stackoverflow.com/questions/58769497/getting-400-bad-request-with-my-firebase-url

I copied and searched these errors into post man, hoping to find what was wrong.
It returned "error": "Index not defined, add ".indexOn": "boards_Id", for path "/pins", to the rules", but my rules were correct. 
So I re-wrote my rules and that didn't fix anything. 
Rob tried to help me figure this out, and he wasn't sure what was happening bc my rules and code seemed to be correct. 

https://github.com/BenJarrett/ASSIGNMENT-Firebase-Pinterest/tree/NEWEST-create-boards

**Comment by drteresavasquez** [comment]
@BenJarrett did you figure this one out?

**Reply by BenJarrett** [reply]
Unfortunately, no I haven’t

**Reply by drteresavasquez** [reply]
@BenJarrett In your rules you have board_id. The error is reading board_Id. You need to update your rules with a capital I.

**Reply by BenJarrett** [reply]
Wow. Can’t believe I missed That.

I will back at my house to start working on finishing in just a little bit.

I will try that and update you on the result!

**Reply by BenJarrett** [reply]
Okay, so I am no longer getting this error, but my pins are not being added. Nothing gets added into firebase as well. 
Every time I click to see all pins on a board and submit a pin, I get a black page with my nav bar. 

But that is progress. Thank you!

I am curious as to why it works when the json and firebase data has it as boards_id without an uppercase?

**Reply by drteresavasquez** [reply]
Check your code. You probably have the uppercase in there.

**Reply by drteresavasquez** [reply]
https://github.com/BenJarrett/ASSIGNMENT-Firebase-Pinterest/blob/NEWEST-create-boards/src/javascripts/helpers/data/pinsData.js#L20

**Reply by BenJarrett** [reply]
Ahh! Thank you! 

Okay, I changed that. Now, when I submit my pin, it does not add a pin to Firebase. So that means I most likely have an issue in my function that is supposed to create pins, correct? 

Also, when I submit a pin, it shows me the pins associated with the board I selected on the add pins form. 

I still cannot view when I clicked the title of my board. It is showing a blank page with my nav. It is not showing me the message I have set up to say "You have no pins" inside my show pins function when a user has no pins associated with that board.

**Reply by drteresavasquez** [reply]
Ok. Focus on the team project for now but you really need to try and figure all this out yourself. My telling you each step is not going to help you grow but come back to it after the team project.

**Reply by BenJarrett** [reply]
Got it. 

And you're right. I've got this. 


Thank you so much for all the help!

**Reply by drteresavasquez** [reply]
Of course! You’re a great dev already. The small sneaky stuff is what gets us. E will get more into it in the tutoring sessions.

**Comment by drteresavasquez** [comment, marked as answer]
@BenJarrett In your rules you have board_id. The error is reading board_Id. You need to update your rules with a capital I.
